core/cas-server-core-services/src/main/java/org/apereo/cas/services/AbstractResourceBasedServiceRegistryDao.java

```diff
@@ -48,12 +51,16 @@
         LOG_SERVICE_DUPLICATE.accept(s2);
         return s1;
     };
+    private static final int INTERVAL = 0;
 
     /**
      * The Service registry directory.
      */
     protected Path serviceRegistryDirectory;
 
+    // PLEASE REVIEW: make sense to this publisher to live in a where ApplicationEventPublisher lives?? and just reference to it
+    private Consumer<AbstractCasEvent> publish = this::publishEvent;
```

Comment:
I mean that by this approach there is no need to pass ```applicationEventPublisher``` all around. Just use the publish consumer that expect an ```AbstractCasEvent``` or even better a ```Supplier<? extends AbstractCasEvent>```.

If this makes sense, this publish consumer should live somewhere accessible from everywhere